### PR TITLE
Feature tmc#219 bugfix

### PR DESCRIPTION
**Pull request by rphh**

### CODE REVIEW

File: sdl2promela/builtins.py [hunk 1]
```diff
@@ -213,81 +214,40 @@ def translate_builtin(
...
-        return (
-            CallBuilder()
-            .withTarget(inlineName)
-            .withParameter(left)
-            .withParameter(__translate_present(call, parameters))
-            .build()
+    right = translate_builtin(call, parameters, types, allTypes)
+    if right is None:
+        raise Exception(
+            f"Builtin {call.name} cannot be used on the left side of assignment"
```
Comment (Lurkerpas) [inline sdl2promela/builtins.py:242]: if right is none then left cannot be used?

File: sdl2promela/translator.py [hunk 2]
```diff
@@ -172,8 +172,28 @@ def __type_name(datatype: Any) -> str:
...
+    if datatype.kind in ["ReferenceType", "EnumeratedType"]:
+        return True
+    return False
+
+
+def __get_assign_value_inline_name(datatype: Any) -> str:
+    """
+    Build name for the inline to assign value to variable with type `datatype`.
+
+    :param datatype: Datatype of varialbe on left side.
```
Comment (Lurkerpas) [inline sdl2promela/translator.py:193]: variable; inline for value assignment

File: sdl2promela/translator.py [hunk 3]
```diff
@@ -332,6 +352,32 @@ def __get_constant_name(
         return candidates[0]
 
 
+def __build_assignment(
+    left: promelamodel.VariableReference, right: promelamodel.Expression, left_type: Any
+) -> promelamodel.Statement:
+    """
+    Build assignment where ``left` is target and `right` is source.
+    The `left_type` is type which determinates if the assignment
```
Comment (Lurkerpas) [inline sdl2promela/translator.py:360]: determines

File: sdl2promela/translator.py (hunk 2 above)
Comment (rphh) [reply sdl2promela/translator.py]: changed

File: sdl2promela/translator.py [hunk 4]
```diff
@@ -172,8 +172,28 @@ def __type_name(datatype: Any) -> str:
     raise NotImplementedError(f"__type_name is not implemented for {datatype.kind}")
 
 
-def __get_assign_value_inline_name(type: Any) -> str:
-    return "{}_assign_value".format(__type_name(type))
+def __requires_assign_value_inline(datatype: Any) -> bool:
+    """
+    Checks if the assignment to variable of type `datatype`:
+    requires call of inline or normal assignment operator.
+    Types defined in ASN.1 dataview model requires inline call.
```
[the remark this answers is not in the document]
Comment (rphh) [reply sdl2promela/translator.py:179]: changed

File: sdl2promela/builtins.py (hunk 1 above)
Comment (rphh) [reply sdl2promela/builtins.py]: changed

File: tests/unit/translator/resources/for_each.pml [hunk 5]
```diff
@@ -12,42 +12,48 @@ inline Taskwithloop_0_transition(id)
...
-      for(i0 : 0 .. (global_state.taskwithloop.variable_param.length - 1))
       {
-        MyInteger_assign_value(each, global_state.taskwithloop.variable_param.data[i0]);
         MyInteger each;
         int i0;
-      };
+        for(i0 : 0 .. (global_state.taskwithloop.variable_param.length - 1))
+        {
+          MyInteger_assign_value(each, global_state.taskwithloop.variable_param.data[i0]);
+          MyInteger each;
```
[the remark this answers is not in the document]
Comment (Lurkerpas) [reply tests/unit/translator/resources/for_each.pml:21]: OK, but why is it defined twice?

[the remark this answers is not in the document]
Comment (rphh) [reply tests/unit/translator/resources/for_each.pml:21]: Ech, this is old test. The input is `task for each in sequenceOfVariable: -- do something with each`. There are many similar tasks in this process, the opengeode allows that.

[the remark this answers is not in the document]
Comment (rphh) [reply tests/unit/translator/resources/for_each.pml:21]: ok, this is a serious regression

File: sdl2promela/translator.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (rphh) [reply sdl2promela/translator.py]: right, changed

File: tests/unit/translator/resources/for_each.pml (hunk 5 above)
[the remark this answers is not in the document]
Comment (rphh) [reply tests/unit/translator/resources/for_each.pml]: Ok, should be fixed now. There is also integration test to check the behaviour of generated code during execution

**Review by Lurkerpas — APPROVED** [review]

**Pull request merged**